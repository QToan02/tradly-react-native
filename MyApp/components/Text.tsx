import { SizableText, styled } from "tamagui";

const Text = styled(SizableText, {
  name: "CustomText",
  variants: {
    unstyled: {
      false: {
        size: "$2",
        color: "#f21",
      },
    },
  } as const,
  defaultVariants: {
    unstyled: false,
  },
});

export default Text;
