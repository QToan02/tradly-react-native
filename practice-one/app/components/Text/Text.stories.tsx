import { ComponentMeta, ComponentStory } from '@storybook/react-native'

import { Text } from '../index'

export default {
  title: 'Text',
  component: Text,
} as ComponentMeta<typeof Text>

const Template: ComponentStory<typeof Text> = (args) => <Text {...args} />

export const Default = Template.bind({})
Default.args = {
  content: 'My text',
}

export const TextWithSize = Template.bind({})
TextWithSize.args = {
  content: 'My bigger text',
  size: 'lg',
}

export const TextWithCustomStyle = Template.bind({})
TextWithCustomStyle.args = {
  content: 'My text is colorful',
  style: {
    color: 'aqua',
  },
}
