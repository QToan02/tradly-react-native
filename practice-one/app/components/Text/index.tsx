import { Text as NativeText, TextStyle } from 'react-native'
import styles from './styles'

export interface TextProps {
  content: string
  size?: 'sm' | 'md' | 'lg' | 'xl'
  style?: TextStyle | TextStyle[]
}

const Text = ({ content, size = 'sm', style = {} }: TextProps) => {
  return <NativeText style={[styles.text, styles[size], style]}>{content}</NativeText>
}

export default Text
