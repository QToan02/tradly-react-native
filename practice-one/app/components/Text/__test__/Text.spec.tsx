import { render } from '@testing-library/react-native'
import Text from '../index'

test('Render text component', () => {
  const component = render(<Text content="my normal text" />).toJSON()

  expect(component).toMatchSnapshot()
})

test('Text component when rendered have right content as prop', () => {
  const { getByText } = render(<Text content="my normal text" />)

  expect(getByText('my normal text')).toBeTruthy()
})
