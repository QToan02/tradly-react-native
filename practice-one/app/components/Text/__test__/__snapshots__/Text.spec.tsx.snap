// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render text component 1`] = `
<Text
  style={
    [
      {
        "color": "#fff",
        "fontFamily": "Montserrat_400Regular",
      },
      {
        "fontSize": 14,
      },
      {},
    ]
  }
>
  my normal text
</Text>
`;
